app/assets/javascripts/discourse/app/controllers/topic.js

```diff
@@ -1569,10 +1570,7 @@ export default Controller.extend(bufferedProperty("model"), {
   },
 
   deleteTopicModal() {
-    showModal("delete-topic-confirm", {
-      model: this.model,
-      title: "topic.actions.delete",
-    });
+    this.modal.show(DeleteTopicConfirmModal, { model: this.model });
```

Comment:
One pattern I recently started settling on is always wrapping the modal arg in an object and naming it:

```suggestion
    this.modal.show(DeleteTopicConfirmModal, { model: { topic: this.model }});
```

Then in the modal you have more verbose `this.args.model.topic` but you know what you're acting on 😅 (in this particular case it might be obvious based on the modal's name but…)